cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/common/OAuthStaticAccessTokenMacroEvaluator.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.common;
+
+import io.cdap.cdap.api.common.Bytes;
+import io.cdap.cdap.api.macro.InvalidMacroException;
+import io.cdap.cdap.api.macro.MacroEvaluator;
+import io.cdap.cdap.api.security.store.SecureStore;
+import io.cdap.cdap.api.exception.ErrorCategory;
+import io.cdap.cdap.api.exception.ErrorCategory.ErrorCategoryEnum;
+import io.cdap.cdap.proto.id.NamespaceId;
+import com.google.gson.Gson;
+
+/**
+ * A {@link MacroEvaluator} for resolving the {@code ${oauthStaticAccessToken(provider, credentialId)}} macro
+ * function. It fetches the stored access token from the secure store.
+ */
+public class OAuthStaticAccessTokenMacroEvaluator implements MacroEvaluator {
```

Comment:
On a second thought, I don't think we need a new `MacroEvaluator` for this usecase.

In the `getOAuthCredential()` request [here](https://github.com/cdapio/cdap/blob/05432b1589fcc09f31d041f75a859573cc33df1b/cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/OAuthHandler.java#L236C15-L236C33) : we can check if access_token is stored already for that credential and provider, return that access_token otherwise follow the refresh_token flow.

In this way, we can reuse the existing `OAuthAccessTokenMacroEvaluator`.

Thanks @sahusanket for pointing this out.

Reply:
This is what I originally proposed in my design doc but you had requested that I make a new Macro instead. Is there a particular reason for us going back to the original proposal here?